kosat-core/src/commonMain/kotlin/org/kosat/util.kt

```diff
@@ -9,3 +9,27 @@ fun Double.round(decimals: Int): Double {
     repeat(decimals) { multiplier *= 10 }
     return kotlin.math.round(this * multiplier) / multiplier
 }
+
+/**
+ * Takes a list of literals, sorts it and removes duplicates in place,
+ * then checks if the list contains a literal and its negation
+ * and returns true if so.
+ */
+fun sortDedupAndCheckComplimentary(lits: MutableList<Lit>): Boolean {
```

Comment:
`util.kt` is not for core functions. Here, we place general-purpose functions that "should be in STD, but they are not".

`sortDedupAndCheckComplimentary` is definitely a part of "CDCL" algorithm, so place it where it would be more suitable.